lib/Browser.js

```diff
@@ -38,11 +38,19 @@ class Browser extends EventEmitter {
     this._connection.setClosedCallback(() => {
       this.emit(Browser.Events.Disconnected);
     });
+    this._connection.on('error', this._onConnectionError.bind(this));
     this._connection.on('Target.targetCreated', this._targetCreated.bind(this));
     this._connection.on('Target.targetDestroyed', this._targetDestroyed.bind(this));
     this._connection.on('Target.targetInfoChanged', this._targetInfoChanged.bind(this));
   }
 
+  _onConnectionError(err) {
+    // we're not going to call the closeCallback if we closed due to an error
+    if (this.listenerCount(Browser.Events.ConnectionError) === 0) throw err;
```

Comment:
I'm not sure I follow this: what kind of situation does it handle?

Reply:
Well, the default logic of the `EventTarget` is that when there are no subscribers to an event called `error`, an error is thrown. I wanted to replicate that behavior while still allowing users to handle the event.

Reply:
And not call it error.

Reply:
I see. Can we consider connection errors as failures and dispatch "disconnected" event instead? From the clients point of view, this situation is equal to the browser connection becoming non-functional.

Reply:
Alright, that makes sense.

I just realized that there also needs to be some code to clean up the browser process if the WS connection breaks.

Reply:
> I just realized that there also needs to be some code to clean up the browser process if the WS connection breaks.

I believe all the cleanup should happen on the page level, see #1454. But it can be addressed separately.